### DISCUSSION TITLE
comparing tab management for Min, Vivaldi and Opera

### DISCUSSION THREAD

**Discussion by shalva97** [minbrowser/min#1696] [General]
its been few weeks I have been using Vivaldi and Opera. In short Opera seems to be the best in tab management, at least for me.

Vivaldi has just so many options to change and configure, I managed to make my tabs look like this:
![image](https://user-images.githubusercontent.com/22417494/127042021-00428210-45cb-4faa-9c91-75910c473335.png)
and the is really good, except few problems, or few bugs. sometimes tab stack names(task names) are lost, or when I close a tab it may switch to another  tab stack for some reason. 

The only downside was that tab stacks are on the top which is usually not used that much often. I think it would be better if they allowed tabs to be top and tab stacks on the bottom, so the Fit's rule would be more useful for tabs. But who knows, maybe they already added this option and tomorrow it will be released.... there is so many options....

and in case of Opera, things are simple.
![image](https://user-images.githubusercontent.com/22417494/127043189-60b14415-6624-4906-a2cc-7dfc3ddba557.png)
they call them Workspaces, which are on the sidebar, alongside extensions and few other stuff. And I love that, it is just few animated icons, their name shows up when you hover and... thats it.... wont lie here, im on opera but not sure if I will stay here.

they bot have Sync, which promise to save settings and tabs to cloud. In case of Vivaldi, its useless, it does sync settings but all the tabs and tab stacks are missing. Opera's Sync is a bit better, tabs are still missing but workspaces are there. And using Git, I can sync everything Min has....

Min seems to have more potential. drag and drop is already here and works well, but is missing in Opera and is a bit buggy on Vivaldi. Also we have a lot more space on task view, lots more functionality can be added. I wish Min had media control buttons on task view, containers like Firefox had, translations, and a mobile app, and an a ability to make me tea, why not, haha.

**Comment by shalva97** [comment]
yeah, this post is more like "comparing browsers to my own needs", but man I spend so much time writing.... hope it is a bit useful...

**Comment by PalmerAL** [comment]
Yeah, this raises some good points. I do think we need to figure out how to make task switching quicker - with Opera, you can switch in one click, because everything is in the sidebar. In Min, you need to find the menu button, and then find the right task, and then find the tab in the list.

However, I see two issues with the Opera approach (I haven't tried it, just looking at your screenshot): 1) It doesn't look like there's a way to see tabs in a list view, which is useful with lots of tabs, and 2) The task switcher is always taking up space on screen even when you don't want it to.

I wonder about some kind of hybrid design where you hover over the menu button, it shows you a small task list to quickly switch tasks, or you could click on it to see the full task list.


> Min had media control buttons on task view

Can you explain what you would use this for? (I might have already asked this, sorry if I have). Do you have music or something playing from a background task, and you want to pause/unpause it?

> containers like Firefox had

Agree we should do this. Electron makes this pretty easy; we mostly just need to figure out a UI for it.

> translations

Should be happening soon (I know you're already subscribed to that issue, but for anyone else reading this: #1180 / #1701)

> a mobile app

I've thought about this, but there's no way I have time to do that and still work on the desktop version of Min. So probably not happening.

> an a ability to make me tea

Could be a userscript, if you have an internet-connected kettle :)
